int/api/wallet/import.go

```diff
@@ -61,11 +62,19 @@ func (c *wImport) Handle(params operations.MgmtWalletImportParams) middleware.Re
 
 	newWallet, err := wallet.Imported(walletName, privateKey)
 	if err != nil {
-		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
-			&models.Error{
-				Code:    errorCodeWalletCreateNew,
-				Message: err.Error(),
-			})
+		if strings.HasPrefix(err.Error(), "wallet already imported") {
```

Comment:
Comparing the error message is not a good practice. if we modify the error message your code will not work anymore.
Here you must create a custom error in the "imported" function to add an error code that you can compare.
you can see an example here under "Collecting Detailed Information in a Custom Error" section
https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go